### PR TITLE
Updates IAM user model with MHV id, LOA fix

### PR DESCRIPTION
**Pull request by patrickvinograd**
Adds MHV identifier to IAM user identity payload, for use in invoking MHV APIs. 

Also fixes LOA interpretation to match VA.gov's, so that DSLogon or MHV assurance level 2 will be interpreted as LOA3 for authorization purposes.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#0000

### CODE REVIEW

**Review by patrickvinograd — COMMENTED** [review]
Alastair's additions LGTM

**Review by kreek — APPROVED** [review]
LGTM 🚀

File: app/models/iam_user_identity.rb [hunk 1]
```diff
@@ -5,6 +5,9 @@
 # introspect endpoint.Adds IAM sourced versions of ICN, EDIPI, and SEC ID to pass to the IAMUser model.
 #
 class IAMUserIdentity < ::UserIdentity
+  PREMIUM_LOAS = [2, 3].freeze
+  UPGRADE_AUTH_TYPES = %w[DSL MHV].freeze
```
Comment (jjulian) [inline app/models/iam_user_identity.rb:9]: Can these values ever come to us in lower or mixed case?

File: app/models/iam_user_identity.rb [hunk 2]
```diff
@@ -52,4 +59,14 @@ def set_expire
   def uuid
     Digest::UUID.uuid_v5(@iam_sec_id, @iam_icn)
   end
+
+  # Return a single mhv id from a possible comma-separated list value attribute
+  def self.valid_mhv_id(id_from_profile)
+    # TODO: site login fails if multiple _different_ MHV IDs are present
+    # Should this code be doing the same? Or at least logging a warning?
+    # see lib/saml/user_attributes/ssoe.rb for example validation
```
Comment (jjulian) [inline app/models/iam_user_identity.rb:67]: I feel like we should resolve these comments before merging.

**Review by jjulian — APPROVED** [review]
Looks good, just two notes

File: app/models/iam_user_identity.rb (hunk 2 above)
Comment (patrickvinograd) [reply app/models/iam_user_identity.rb:67]: It's not clear  what the right thing to do is. On VA.gov, when this  occurs we prevent sign-in, and we have the opportunity to present a clear  error  message during the sign-in flow. 

Here, IAM is going to issue the access token successfully. We can prevent this API request, but we could be instantiating this user identity object for any of the OAuth-based API endpoints - it's not like there's a first API  request  that always happens after token issuance. That means any part of the app that makes an API request needs to be prepared to handle getting an error back and presenting something meaningful to the user. And we need to define what that error is.

Would you be okay if we filed a launch-blocker ticket and revisited it, in the interest of  getting the MHV integration moving  along?

Comment (jjulian) [reply app/models/iam_user_identity.rb:67]: That sounds good, thanks Patrick. For the sake of this PR, let's log a warning and at least we have some visibility into how often this occurs.

Comment (lexicalninja) [reply app/models/iam_user_identity.rb:67]: Do we want to just log a warning or create a Crashlytics event for this?

Comment (lexicalninja) [reply app/models/iam_user_identity.rb:67]: I guess probably not crashlytics here, but is there a way to make it easy to pull a report on this?

Comment (patrickvinograd) [reply app/models/iam_user_identity.rb:67]: Two other mitigation options:

1) Get the IAM team to alter their logic such that they don't issue an access token if they detect multiple MHV IDs. But what if the  second  MHV ID is added _after_ the refresh token is issued? Complex.

2) Just gate access to MHV features such as SM - i.e. remove the MHV ID from the session, and  thus secure messaging will disappear from the authorizedServices list. There's no reason to gate access to claims becaause of a duplicate  MHV ID, for example.  Might need a "why can't I  access?" UX  consideration though.

Comment (patrickvinograd) [reply app/models/iam_user_identity.rb:67]: I filed this issue for follow-up: https://github.com/department-of-veterans-affairs/va.gov-team/issues/19971

Comment (patrickvinograd) [reply app/models/iam_user_identity.rb:67]: We can also log a sentry event for it for visibility.

**Review by thilton-oddball — APPROVED** [review]
Looks good!

File: app/models/iam_user_identity.rb (hunk 1 above)
Comment (kreek) [reply app/models/iam_user_identity.rb:13]: I'd need to double check with the IAM team, but I don't think so

Comment (patrickvinograd) [reply app/models/iam_user_identity.rb:13]: Anything's possible but I really doubt it.

**Pull request merged**